### PR TITLE
add developer portfolio feature + portfolio clubs

### PR DESCRIPTION
**Pull request by laurosilvacom**
We needed a landing page for portfolio clubs and cards that could serve as CTAs to drive sign-ups. Instead of creating portfolio club data in isolation, this PR leverages the established pattern for creating features. 

The `Craft a Business Oriented Developer Portfolio` feature contains a `Portfolio Clubs` resource, and inside of that, `Clubs` are structured as resources.

**Note:** the developer portfolio feature page still needs further curation, but that's probably a separate PR. 

### club card component
image

### club landing page
image

### developer portfolio feature
image

### developer portfolio page
image

### CODE REVIEW

File: src/pages/learn/developer-portfolio/index.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+          </div>
+        </div>
+      </div>
+    </div>
+  )
+}
+
+export default DeveloperPortfolio
+
+export const developerPortfolioQuery = groq`*[_type == 'resource' && slug.current == "build-business-oriented-portfolio"][0]{
```
Comment (laurosilvacom) [inline src/pages/learn/developer-portfolio/index.tsx:63]: `developerPortfolioQuery` provides all the data for the feature section and the feature page.

File: src/pages/learn/developer-portfolio/index.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+      "slug": slug.current,
+      path,
+      meta,
+      image,
+      summary,
+	}
+}`
+
+export const reduxClubQuery = groq`*[_type == 'resource' && slug.current == "build-business-oriented-portfolio"][0]{
+  "redux": resources[0].resources[slug.current == "redux"][0]{
```
Comment (laurosilvacom) [inline src/pages/learn/developer-portfolio/index.tsx:82]: Isolated queries are generated for each club's landing page.

File: src/pages/clubs/redux/index.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+          />
+          <h1 className="max-w-screen-md lg:text-6xl md:text-5xl sm:text-4xl text-3xl w-full font-extrabold leading-tighter mt-8">
+            {data.redux.title}
+          </h1>
+          <p className="md:text-lg text-base font-semibold mt-4 text-gray-500">
+            Portfolio Project Club ・ {data.redux.subTitle}
+          </p>
+        </div>
+        <Markdown className="prose dark:prose-dark dark:prose-lg-dark prose-lg">
+          {data.redux.description}
```
Comment (laurosilvacom) [inline src/pages/clubs/redux/index.tsx:53]: All the data for the landing page is managed in Sanity. No need for MDX files.

File: src/pages/learn/developer-portfolio/index.tsx [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+  image,
+  "clubs": resources[0].resources[]{
+      title,
+      subTitle,
+      "slug": slug.current,
+      path,
+      meta,
+      image,
+      summary,
+	}
```
Comment (laurosilvacom) [inline src/pages/learn/developer-portfolio/index.tsx:78]: `clubs` are nested inside `Feature Portfolio Clubs`, which is a resource in `Craft a Business Oriented Developer Portfolio`. We can continue adding more resources or clubs without breaking the front end.

File: studio/schemas/documents/resource.js [hunk 5]
```diff
@@ -118,6 +118,10 @@ export default {
             title: 'portfolio',
             value: 'portfolio',
           },
+          {
+            title: 'club',
+            value: 'club',
+          },
```
Comment (laurosilvacom) [inline studio/schemas/documents/resource.js:124]: adds a new type to resources for clubs.

**Comment by dealingwith** [comment]
I made one tiny update to the CTA copy in Sanity. I gave one piece of nitpicky layout feedback IRT to @laurosilvacom 

My somewhat less nitpicky but still nitpicky feedback is IRT to this bit of UI based on @MaggieAppleton's Figma:

Screen Shot 2021-04-22 at 11 35 13 AM

🔴 Live -- feels like a pattern based on live video ala YouTube or Twitch and just throws me off a little bit. Perhaps there should be "Upcoming" and "In Process" statuses instead? The latter might not make sense unless we want to communicate the types and quantity of _stuff_ going on in the egghead community. It probably doesn't make sense to have on the homepage, at least.

"Join" is already a good CTA, so this bit of UI might be superfluous anyway.

Screen Shot 2021-04-22 at 11 40 28 AM
Screen Shot 2021-04-22 at 11 42 27 AM

**Comment by laurosilvacom** [comment]
@dealingwith makes a good point. The "🔴 Live" is confusing from a functionality standpoint. 

The updated card:

image

File: src/pages/learn/developer-portfolio/index.tsx [hunk 6]
```diff
@@ -0,0 +1,110 @@
+import * as React from 'react'
+import groq from 'groq'
+import {sanityClient} from 'utils/sanity-client'
+import Image from 'next/image'
+import Link from 'next/link'
+import {track} from 'utils/analytics'
+import Markdown from 'react-markdown'
+
+const DeveloperPortfolio: React.FC<any> = ({data}) => {
```
[the remark this answers is not in the document]
Comment (laurosilvacom) [reply src/pages/learn/developer-portfolio/index.tsx:10]: Updated! 

This page requires further curation. Having Parker's [article](https://egghead.io/blog/saas-application-with-next-js-stripe-and-firebase) on this page would be great, along with other resources. Especially if we plan to use the article component https://github.com/eggheadio/egghead-next/pull/580

image

**Review by zacjones93 — APPROVED** [review]

**Pull request merged**